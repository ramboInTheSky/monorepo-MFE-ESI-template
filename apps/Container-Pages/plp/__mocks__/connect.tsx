import React from "react"
import {Provider} from "react-redux"
import {render} from "@testing-library/react"
import {mockState, mockConfigureStore} from "./mockStore"

interface AnyObject {
    [key: string]: any
}

interface MockComponentProps {
    propsRef: {current: any}
    props: AnyObject
}

const MockComponent = jest.fn(({propsRef, ...props}: MockComponentProps) => {
    // eslint-disable-next-line no-param-reassign
    propsRef.current = props
    return null
}) as any

export function renderConnect(connect: Function, state: AnyObject = {}, otherProps: Record<any, any> = {}) {
    const props = React.createRef()
    const store = mockConfigureStore({...mockState, ...state})
    const ConnectedMock = connect(MockComponent)
    render(
        <Provider store={store}>
            <ConnectedMock {...otherProps} propsRef={props} />
        </Provider>,
    )
    return props.current
}

export function createMockStateForTabbedFilters(searchState: Record<any, any> = {}) {
    return {
        ...mockState,
        tabbedFilters: {
            ...mockState.tabbedFilters,
            ...searchState,
        },
    }
}
