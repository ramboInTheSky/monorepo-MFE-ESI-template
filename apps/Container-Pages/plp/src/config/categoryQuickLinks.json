{
  "amido": {
    "uk": [
      {
        "href": "{SITE_URL}/women",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/women.jpg",
        "title": "Women",
        "description": "From dresses to denim, browse catwalk inspired trends from this season’s women’s collection",
        "imageAlt": "women"
      },
      {
        "href": "{SITE_URL}/men",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/mens.jpg",
        "title": "Men",
        "description": "Be on trend with the latest men’s shirts, tops, trousers and jeans from this men’s clothing range.",
        "imageAlt": "men"
      },
      {
        "href": "{SITE_URL}/girls",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/girls.jpg",
        "title": "Girls",
        "description": "Gorgeous girls’ dresses and cool coloured jeans make up this fun and fashionable clothing range.",
        "imageAlt": "girls"
      },
      {
        "href": "{SITE_URL}/boys",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/boys.jpg",
        "title": "Boys",
        "description": "Mix and match these boys’ clothing and shoe collections for boys fashion he’ll love.",
        "imageAlt": "boys"
      },
      {
        "href": "{SITE_URL}/branded",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/brands.jpg",
        "title": "Designer Brands",
        "description": "Explore A-Z of the latest designer brands at Amido, with great delivery options.",
        "imageAlt": "designer brands"
      },
      {
        "href": "{SITE_URL}/shoes",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/shoes.jpg",
        "title": "Shoes",
        "description": "With our latest footwear collections you will always have your fashion foot forward.",
        "imageAlt": "shoes"
      },
      {
        "href": "{SITE_URL}/homeware",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/homeware.jpg",
        "title": "Homeware",
        "description": "Get that welcome home feel with stunning furniture collections with 5 day delivery.",
        "imageAlt": "homeware"
      }
    ],
    "international": [
      {
        "href": "{SITE_URL}/women",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/women.jpg",
        "mpStart":"<!-- mp_trans_enable_start -->",
        "title": "Women",
        "description": "From dresses to denim, browse catwalk inspired trends from this season’s women’s collection",
        "imageAlt": "women",
        "mpEnd":"<!-- mp_trans_enable_end -->"
      },
      {
        "href": "{SITE_URL}/men",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/mens.jpg",
        "mpStart":"<!-- mp_trans_enable_start -->",
        "title": "Men",
        "description": "Be on trend with the latest men’s shirts, tops, trousers and jeans from this men’s clothing range.",
        "imageAlt": "men",
        "mpEnd":"<!-- mp_trans_enable_end -->"
      },
      {
        "href": "{SITE_URL}/girls",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/girls.jpg", 
        "mpStart":"<!-- mp_trans_enable_start -->",
        "title": "Girls",
        "description": "Gorgeous girls’ dresses and cool coloured jeans make up this fun and fashionable clothing range.",
        "imageAlt": "girls",
        "mpEnd":"<!-- mp_trans_enable_end -->"
      },
      {
        "href": "{SITE_URL}/boys",
        "imageSrc": "{CDN_URL}/common/items/default/default/Sitemaps/X49/Departments/boys.jpg",
        "mpStart":"<!-- mp_trans_enable_start -->",
        "title": "Boys",
        "description": "Mix and match these boys’ clothing and shoe collections for boys fashion he’ll love.",
        "imageAlt": "boys",
        "mpEnd":"<!-- mp_trans_enable_end -->"
      }
    ]
  }
}
