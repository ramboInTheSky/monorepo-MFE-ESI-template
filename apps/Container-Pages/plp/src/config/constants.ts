export const BASELINECSS_MATERIALUI = "materialui"
export const BASELINECSS_CUSTOM = "custom"
export const JSS_SERVERSIDE = "plp-jss-server-side-"

export enum Horizontal {
    ltr = "left",
    rtl = "right",
}

export type FilterType = "filter" | "price" | "feat"

export const FILTER_SETTINGS_COOKIE = "filterSettings"

export enum SearchApiRequestTypes {
    Category = "Category",
    Keyword = "Keyword",
}

export const SEARCH_CATEGORY_SEGMENT = "shop"

export enum ViewLinkOptions {
    none,
    viewMore,
    viewAll,
}

export const FILTERS_SHOW_VIEW_ALL_COUNT = 20
export const FILTERS_SHOW_VIEW_MORE_COUNT = 8
export const FILTERS_SHOW_VIEW_MORE_COUNT_EXTENDED = 30

export const REALM_HEADER = "x-monorepo-realm"
export const TERRITORY_HEADER = "x-monorepo-territory"
export const LANGUAGE_HEADER = "x-monorepo-language"
export const DEV_ESI_HEADER = "test-with-local-esi"
export const VIEWPORT_SIZE_HEADER = "x-monorepo-viewport-size"
export const TIME_MACHINE_DATE_HEADER = "x-monorepo-time-machine-date"
export const SEARCH_BANNER_HEADER = "x-monorepo-banner-format"
export const SEARCH_BANNER_HEADER_DEFAULT = "html"
export const LOCAL_STORAGE_ACTIVE_DEPT_NAME = "PrimaryNavDeptSelector"
export const GTM_SEARCH_RESULTS = "searchResults"

export const FILTER_GRID_SCROLL_HEIGHT = 690
export const FILTER_GRID_TOP_SPACER_HEIGHT = 25

export const DIGIT_REGEX = RegExp(/^\d/)

export const HISTORY_EVENT = "PLP-FILTER-EVENT"

export const STICKY_HEADER_SCROLL_TRIGGER = 44

export const TOKEN_CANCELLATION_FLAG = "TOKEN_CANCELLATION"

export const TERRITORY_GB = "gb"

export const REGION_UK = "uk"

export const REGION_INTERNATIONAL = "international"

export const LANGUAGE_ENGLISH = "en"

export const DEFAULT_FILTER_SPINNER_TIMEOUT = 5000

export const BRAND = "brand"
export const BRAND_0_9 = "0-9"

export const GENDER = "gender"

export const CLEARANCE = "Clearance"

export enum ESISeparator {
    ProductFragment = "<=ProductFragment=>",
    SearchBanner = "<=SearchBanner=>",
}
export const FOOTER_ID = "footer-entrypoint"
export const PLP_ID = "plp-entrypoint"

export const FACETS_CONTAINER_OFFSET = 25

export const SEARCH_BANNER_INCLUDE_COMPONENTS = "search-banner"
export const SEO_METADATA_INCLUDE_COMPONENTS = "seo-metadata"

export const CACHE_HEADERS_CACHE_CONTROL = "cache-control"
export const CACHE_HEADERS_EDGE_CACHE_TAG = "edge-cache-tag"
export const CACHE_HEADERS_LAST_MODIFIED = "last-modified"
export const SEARCH_BANNER_CACHE_TAG = "searchbannerui"
export const SEARCH_CACHE_TAG = "plpui"

export const ITEM_NUMBER_REGEX = RegExp("^\\s?\\w\\d{2}-?\\d{3}\\s?$")
export const BR_UID_2 = "_br_uid_2"
export const BR_MT_SEARCH = "_br_mt_search"
export const BR_UID_DEFAULT_VALUE = "uid%3D9624001141133%3Av%3D13.0%3Ats%3D1578565383488%3Ahc%3D138"
export const CATEGORY_FILTERS_REGEX = /(([a-z, 0-9]*).-[a-z, 0-9,%]*)+/g

export const GTM_PRODUCT_IMPRESSIONS_EVENT = "ee-productImpression"
export const SWIPER_ITEMS_PER_PAGE = 4

export const featAvailable = "feat:available"
export const featDeliverBy = "feat:deliveryby"
export const featBackInStock = "feat:backinstock"
