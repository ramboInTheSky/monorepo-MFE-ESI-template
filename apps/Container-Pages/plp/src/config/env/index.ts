const getLocalURL = localPort =>
    INTERNAL.REACT_APP_LOCAL || INTERNAL.DEVELOPMENT ? `http://localhost:${localPort}` : ""

const INTERNAL = Object.freeze({
    PRODUCTION: process.env.NODE_ENV === "production",
    DEVELOPMENT: process.env.NODE_ENV === "development",
    REACT_APP_LOCAL: process.env.REACT_APP_LOCAL,
    REACT_APP_BLOB_STORAGE_PATH: "/static-content",
    REACT_APP_INCLUDE_VENDORS: process.env.REACT_APP_INCLUDE_VENDORS || false,
    REACT_APP_ENABLE_CYPRESS_SETTINGS: process.env.REACT_APP_ENABLE_CYPRESS_SETTINGS || false,
})

const ENV_VARS = Object.freeze({
    REACT_APP_BLOB_STORAGE_SSR_BASEURL: process.env.REACT_APP_BLOB_STORAGE_SSR_BASEURL || "http://localhost:3000",
    PORT: process.env.PORT || 3000,
    NODE_ENV: process.env.NODE_ENV,
    REACT_APP_CDN_BASEURL: process.env.REACT_APP_CDN_BASEURL || "https://xcdn.amido.com/content",
    REACT_APP_API_BASEURL_SEARCH: process.env.REACT_APP_API_BASEURL_SEARCH || "https://sx-test.amido.com/api/search",
    REACT_APP_API_BASEURL_PRODUCT_SUMMARY:
        process.env.REACT_APP_API_BASEURL_PRODUCT_SUMMARY || "https://sx-test.amido.com/api/product-summary",
    REACT_APP_API_BASEURL_SEARCH_BANNER: process.env.REACT_APP_API_BASEURL_SEARCH_BANNER || getLocalURL(3339),

    POD_NAME: process.env.POD_NAME,
    REACT_APP_SERVE_PATH_PREFIX: process.env.REACT_APP_SERVE_PATH_PREFIX || "",
    REACT_APP_APP_URL:
        process.env.REACT_APP_APP_URL ||
        (INTERNAL.REACT_APP_LOCAL || INTERNAL.DEVELOPMENT ? "http://localhost:3009" : ""),
    REACT_APP_HEADER_BASEURL: process.env.REACT_APP_HEADER_BASEURL || getLocalURL(3004),
    REACT_APP_HEADER_ASSETS_PATH: process.env.REACT_APP_HEADER_ASSETS_PATH || "/headerstatic",

    REACT_APP_MEGANAV_BASEURL: process.env.REACT_APP_MEGANAV_BASEURL || getLocalURL(3005),

    REACT_APP_FOOTER_BASEURL: process.env.REACT_APP_FOOTER_BASEURL || getLocalURL(3002),
    REACT_APP_FOOTER_ASSETS_PATH: process.env.REACT_APP_FOOTER_ASSETS_PATH || "/footerstatic",

    REACT_APP_PROD_SUMM_BASEURL: process.env.REACT_APP_PROD_SUMM_BASEURL || getLocalURL(3001),
    REACT_APP_PROD_SUMM_ASSETS_PATH: process.env.REACT_APP_PROD_SUMM_ASSETS_PATH || "/productsummarystatic",

    REACT_APP_USE_TIME_MACHINE_COOKIE: process.env.REACT_APP_USE_TIME_MACHINE_COOKIE || "false",

    ASSETS_PATH: process.env.ASSETS_PATH || "/plpstatic",
    LOG_LEVEL: process.env.LOG_LEVEL,
    FNC_NAME: process.env.FNC_NAME,
    REACT_APP_APPINSIGHTS_KEY: process.env.REACT_APP_APPINSIGHTS_KEY,
    ENVIRONMENT_NAME: process.env.ENVIRONMENT_NAME,
    USE_LOCAL_STATIC_CONTENT: process.env.USE_LOCAL_STATIC_CONTENT || "true",
})

const VARS = {...INTERNAL, ...ENV_VARS}

export default VARS
