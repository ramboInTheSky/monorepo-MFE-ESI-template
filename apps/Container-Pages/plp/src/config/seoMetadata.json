{
  "amido": {
    "uk": {
      "category": {
        "withFilters": {
          "title": "Buy {Filters} {SearchTerm} Online | Amido Clothing UK",
          "description": "Shop a wide range of {Filters} {SearchTerm} with Amido Clothing UK. Choose from 1000s of products. Order {Filters} {SearchTerm} now with express delivery.",
          "keywords": "kids designer clothes, luxury childrenswear, amido clothing",
          "robots": ""
        },
        "withoutFilters": {
          "title": "Buy {SearchTerm} Online | Amido Clothing UK",
          "description": "Shop a wide range of {SearchTerm} with Amido Clothing UK. Choose from 1000s of products here. Order {SearchTerm} now with express delivery.",
          "keywords": "kids designer clothes, luxury childrenswear, amido clothing",
          "robots": ""
        }
      },
      "keyword": {
        "withFilters": {
          "title": "Buy {SearchTerm} in {Filters} from the Amido Clothing UK online shop",
          "description": "Shop for {SearchTerm} in {Filters} at amidoclothing.co.uk. Amido day delivery and free returns available. 1000s of products online. Buy {SearchTerm} now!",
          "keywords": "amidoclothing,amidoclothing.co.uk,childrens",
          "robots": ""
        },
        "withoutFilters": {
          "title": "Buy {SearchTerm} from the Amido UK online shop",
          "description": "Shop for {SearchTerm} at amidoclothing.co.uk. Amido day delivery and free returns available. 1000s of products online. Buy {SearchTerm} now!",
          "keywords": "amidoclothing,amidoclothing.co.uk,childrens",
          "robots": ""
        }
      }
    },
    "international": {
      "category": {
        "withFilters": {
          "mpStart": "<!-- mp_trans_enable_start -->",
          "title": "Buy {Filters} {SearchTerm} Online | Amido Clothing {CountryName}",
          "description": "Shop a wide range of {Filters} {SearchTerm} with Amido Clothing {CountryName} now with express delivery.",
          "keywords": "kids designer clothes, luxury childrenswear, amido clothing",
          "robots": "",
          "mpEnd": "<!-- mp_trans_enable_end -->"
        },
        "withoutFilters": {
          "mpStart": "<!-- mp_trans_enable_start -->",
          "title": "Buy {SearchTerm} Online | Amido Clothing {CountryName}",
          "description": "Shop a wide range of {SearchTerm} with Amido Clothing {CountryName} now with express delivery.",
          "keywords": "kids designer clothes, luxury childrenswear, amido clothing",
          "robots": "",
          "mpEnd": "<!-- mp_trans_enable_end -->"
        }
      },
      "keyword": {
        "withFilters": {
          "mpStart": "<!-- mp_trans_enable_start -->",
          "title": "Buy {SearchTerm} in {Filters} from | Amido Clothing {CountryName}",
          "description": "Shop for {SearchTerm} in {Filters} at - Amido Clothing {CountryName}. International shipping and returns available. Buy {SearchTerm} in {Filters} now!",
          "keywords": "",
          "robots": "",
          "mpEnd": "<!-- mp_trans_enable_end -->"
        },
        "withoutFilters": {
          "mpStart": "<!-- mp_trans_enable_start -->",
          "title": "Buy {SearchTerm} from | Amido Clothing {CountryName}",
          "description": "Shop for {SearchTerm} at - Amido Clothing {CountryName}. International shipping and returns available. Buy now!",
          "keywords": "",
          "robots": "",
          "mpEnd": "<!-- mp_trans_enable_end -->"
        }
      }
    }
  }
}
