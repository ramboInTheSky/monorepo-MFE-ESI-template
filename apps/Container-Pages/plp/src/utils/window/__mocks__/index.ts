import {
    getWindow,
    getDocumentScrollTop,
    removeDocumentScrollListener,
    addDocumentScrollListener,
    isSomewhatInViewport,
    isAboveViewport,
    isBelowViewport,
    IS_BROWSER,
} from ".."

jest.mock("..", () => ({
    getWindow: jest.fn(() => ({})),
    IS_BROWSER: jest.fn(() => true),
    isAboveViewport: jest.fn(),
    isBelowViewport: jest.fn(),
    isSomewhatInViewport: jest.fn(),
    getDocumentScrollTop: jest.fn(),
    addDocumentScrollListener: jest.fn(),
    removeDocumentScrollListener: jest.fn(),
}))

function mockListener(fn: jest.Mock) {
    const api = {
        listener: () => null,
    }
    fn.mockImplementation(listener => {
        api.listener = listener
    })
    return api
}

export function mockGetWindow(value: any = window) {
    ;(getWindow as jest.Mock).mockReturnValue(value)
}

export function mockGetDocumentScrollTop(value: number) {
    ;(getDocumentScrollTop as jest.Mock).mockReturnValue(value)
}

export function mockRemoveDocumentScrollListener() {
    return mockListener(removeDocumentScrollListener as jest.Mock)
}

export function mockAddDocumentScrollListener() {
    return mockListener(addDocumentScrollListener as jest.Mock)
}

export function mockIsSomewhatInViewport(result: boolean) {
    ;(isSomewhatInViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsAboveViewport(result: boolean) {
    ;(isAboveViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsBelowViewport(result: boolean) {
    ;(isBelowViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsBrowser(result: boolean) {
    ;(IS_BROWSER as jest.Mock).mockReturnValue(result)
}
