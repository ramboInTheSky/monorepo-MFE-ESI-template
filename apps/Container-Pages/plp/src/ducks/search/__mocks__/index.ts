import {mockState} from "../../../../__mocks__/mockStore"

import {selectHasNextPage, selectHasPreviousPage} from ".."

import {selectFetchTriggerOffset, selectSubsequentItemsPerPage} from "../utils"

jest.mock("../utils", () => ({
    selectFetchTriggerOffset: jest.fn(),
    selectSubsequentItemsPerPage: jest.fn(),
}))
jest.mock("..", () => ({
    fixFacets: jest.fn(() => ({type: "FIX_FACETS"})),
    unfixFacets: jest.fn(() => ({type: "UNFIX_FACETS"})),
    selectItems: jest.fn(),
    selectHasNextPage: jest.fn(),
    selectHasPreviousPage: jest.fn(),
    selectCurrentEndPage: jest.fn(),
    selectCurrentStartPage: jest.fn(),
    selectTotalItemsPerPage: jest.fn(),
    fetchNextPageItemsThunk: jest.fn(() => ({type: "foo"})),
    fetchPreviousPageItemsThunk: jest.fn(() => ({type: "foo"})),
}))

export function mockSelectHasNextPage(result) {
    ;(selectHasNextPage as jest.Mock).mockReturnValue(result)
}

export function mockSelectHasPreviousPage(result) {
    ;(selectHasPreviousPage as jest.Mock).mockReturnValue(result)
}

export function mockSelectFetchTriggerOffset(result) {
    ;(selectFetchTriggerOffset as jest.Mock).mockReturnValue(result)
}
export function mockSelectSubsequentItemsPerPage(result) {
    ;(selectSubsequentItemsPerPage as jest.Mock).mockReturnValue(result)
}

export function createMockStateForSearch(searchState: Record<any, any> = {}) {
    return {
        ...mockState,
        search: {
            ...mockState.search,
            ...searchState,
        },
    }
}
