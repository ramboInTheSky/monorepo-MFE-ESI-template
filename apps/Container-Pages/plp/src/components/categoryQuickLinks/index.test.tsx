import React from "react"
import useMediaQuery from "@mui/material/useMediaQuery"
import {Swiper} from "swiper/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {breakpoints, mockTheme} from "@monorepo/themes"
import {render, screen} from "@testing-library/react"
import {CategoryQuickLinks} from "."
import { mockText } from "../../../__mocks__/mockStore"
import { SWIPER_ITEMS_PER_PAGE } from "../../config/constants"

jest.mock("swiper/react", () => ({
    Swiper: jest.fn(({children}) => <div className="swiper">{children}</div>),
    SwiperSlide: jest.fn(({children}) => <div className="swiper-slide">{children}</div>),
}))

jest.mock("@mui/material/useMediaQuery", () => ({
    __esModule: true,
    default: jest.fn(),
}))

const mockItems = (length) => {
    return Array(length).fill(
        {
            href: "",
            title: "Women",
            description: "Womens description...",
            imageSrc: "/some/image/src.jpg",
            imageAlt: "women",
        }
    // href is used as key in render so needs to be unique
    ).map((item, index) => ({...item, href: `/women/${index}`}))
}

interface Scenario {
    minBreakpoint?: number
}

function mockScenario({minBreakpoint = breakpoints.values.xs}: Scenario) {
    ;(useMediaQuery as jest.Mock).mockImplementation(query => {
        if (query === `(min-width: ${minBreakpoint}px)`) {
            return true
        }
        return false
    })
}

const the = {
    categoryQuickLinks: () => {
        return screen.queryByTestId("plp-category-quick-links")
    },
}

describe("Given <CategoryQuickLinks />", () => {
    afterEach(() => {
        jest.clearAllMocks()
    })

    describe("by default", () => {
        it("should match the snapshot", () => {
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <CategoryQuickLinks items={mockItems(2)} text={mockText} />
                </SCThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe("when there are no items", () => {
        it("should render nothing", () => {
            render(
                <SCThemeProvider theme={mockTheme}>
                    <CategoryQuickLinks items={[]} text={mockText} />
                </SCThemeProvider>,
            )
            expect(the.categoryQuickLinks()).not.toBeInTheDocument()
        })
    })

    describe("when in a medium device and above", () => {
        it("should render a gap of 32px between the swiper slides", () => {
            mockScenario({minBreakpoint: breakpoints.values.md})
            render(
                <SCThemeProvider theme={mockTheme}>
                    <CategoryQuickLinks items={mockItems(2)} text={mockText}/>
                </SCThemeProvider>,
            )
            expect(Swiper).toHaveBeenCalledWith(expect.objectContaining({spaceBetween: 32}), {})
        })
    })

    describe("when in a device that is below medium", () => {
        it("should render a gap of 16px between the swiper slides", () => {
            mockScenario({minBreakpoint: breakpoints.values.sm})
            render(
                <SCThemeProvider theme={mockTheme}>
                    <CategoryQuickLinks items={mockItems(2)} text={mockText} />
                </SCThemeProvider>,
            )
            expect(Swiper).toHaveBeenCalledWith(expect.objectContaining({spaceBetween: 16}), {})
        })
    })

    describe("when there are SWIPER_ITEMS_PER_PAGE items or less", () => {
        it("should set navigation to false", () => {
            render(
                <SCThemeProvider theme={mockTheme}>
                    <CategoryQuickLinks items={mockItems(SWIPER_ITEMS_PER_PAGE)} text={mockText} />
                </SCThemeProvider>,
            )
            expect(Swiper).toHaveBeenCalledWith(expect.objectContaining({navigation: false}), {})
        })
    })

    describe("when there are more than SWIPER_ITEMS_PER_PAGE items", () => {
        it("should set navigation to true", () => {
            render(
                <SCThemeProvider theme={mockTheme}>
                    <CategoryQuickLinks items={mockItems(SWIPER_ITEMS_PER_PAGE + 1)} text={mockText} />
                </SCThemeProvider>,
            )
            expect(Swiper).toHaveBeenCalledWith(expect.objectContaining({navigation: true}), {})
        })
    })
})
