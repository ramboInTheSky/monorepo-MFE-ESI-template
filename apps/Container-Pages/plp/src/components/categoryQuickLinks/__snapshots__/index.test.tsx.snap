// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <CategoryQuickLinks /> by default should match the snapshot 1`] = `
<DocumentFragment>
  .c3 {
  width: 7.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  width: 7.5rem;
  height: 11.25rem;
}

.c5 {
  width: 100%;
}

.c6 {
  font-size: 0.875rem;
  font-family: "AzoSansRegular","Helvetica Neue",Arial,sans-serif;
  line-height: 1.3125rem;
  -webkit-letter-spacing: 0.0075rem;
  -moz-letter-spacing: 0.0075rem;
  -ms-letter-spacing: 0.0075rem;
  letter-spacing: 0.0075rem;
  margin-top: 0.625rem;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  width: 100%;
}

.c0 >:not(:first-child) {
  margin-top: 0.875rem;
}

.c0 .swiper-container {
  margin-left: 0;
  margin-right: 0;
}

.c0 .swiper-wrapper {
  height: auto;
}

.c0 .swiper-slide {
  width: 7.5rem;
}

.c0 .swiper-slide:last-child {
  margin-right: 0 !important;
}

.c0 .swiper-button-prev,
.c0 .swiper-button-next {
  color: #000;
  width: 1.75rem;
  height: 2.25rem;
  background: rgba(255,255,255,1);
  display: none;
  top: 38%;
}

.c0 .swiper-button-prev.swiper-button-disabled,
.c0 .swiper-button-next.swiper-button-disabled {
  display: none;
}

.c0 .swiper-button-prev:after,
.c0 .swiper-button-next:after {
  font-size: 0.875rem;
  font-weight: bold;
}

.c0 .swiper-button-prev {
  left: 0;
  border-left: 0;
  border-radius: 0 0.5rem 0.5rem 0;
}

.c0 .swiper-button-prev:after {
  margin-left: -0.125rem;
}

.c0 .swiper-button-next {
  right: 0;
  border-right: 0;
  border-radius: 0.5rem 0 0 0.5rem;
}

.c0 .swiper-button-next:after {
  margin-right: -0.125rem;
}

.c1 {
  font-size: 1rem;
  font-family: "AzoSansMedium","Helvetica Neue",Arial,sans-serif;
  font-weight: 500;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.0075rem;
  -moz-letter-spacing: 0.0075rem;
  -ms-letter-spacing: 0.0075rem;
  letter-spacing: 0.0075rem;
}

.c2 {
  width: 100%;
  overflow-x: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:768px) {
  .c3 {
    width: 11.25rem;
  }
}

@media (min-width:768px) {
  .c4 {
    width: 11.25rem;
    height: 16.875rem;
  }
}

@media (min-width:768px) {
  .c0 .swiper-slide {
    width: 11.25rem;
  }
}

@media (min-width:1024px) {
  .c0 .swiper-button-prev,
  .c0 .swiper-button-next {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media all and (-ms-high-contrast:none) {
  .c0 .swiper-button-prev,
  .c0 .swiper-button-next {
    position: absolute;
    top: 36%;
    width: 1.6875rem;
    margin-top: -1.375rem;
    z-index: 10;
    cursor: pointer;
    background-size: 1.25rem 1rem;
    background-position: center;
    background-repeat: no-repeat;
  }

  .c0 .swiper-button-prev.swiper-button-disabled,
  .c0 .swiper-button-next.swiper-button-disabled {
    opacity: 0.35;
    cursor: auto;
    pointer-events: none;
  }

  .c0 .swiper-button-prev,
  .c0 .swiper-container-rtl .swiper-button-next {
    background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg'%20viewBox%3D'0%200%2027%2044'%3E%3Cpath%20d%3D'M0%2C22L22%2C0l2.1%2C2.1L4.2%2C22l19.9%2C19.9L22%2C44L0%2C22L0%2C22L0%2C22z'%20fill%3D'%23000000'%2F%3E%3C%2Fsvg%3E");
    left: -0.0625rem;
    right: auto;
  }

  .c0 .swiper-button-next,
  .c0 .swiper-container-rtl .swiper-button-prev {
    background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg'%20viewBox%3D'0%200%2027%2044'%3E%3Cpath%20d%3D'M27%2C22L27%2C22L5%2C44l-2.1-2.1L22.8%2C22L2.9%2C2.1L5%2C0L27%2C22L27%2C22z'%20fill%3D'%23000000'%2F%3E%3C%2Fsvg%3E");
    right: -0.0625rem;
    left: auto;
  }
}

<div
    class="c0"
    data-testid="plp-category-quick-links"
  >
    <h1
      class="MuiTypography-root c1 MuiTypography-body1"
    >
      Shop Now
    </h1>
    <div
      class="c2"
    >
      <div
        class="swiper"
      >
        <div
          class="swiper-slide"
        >
          <a
            class="c3"
            data-testid="plp-category-quick-link-item"
            href="/women/0"
          >
            <div
              class="c4"
            >
              <img
                alt="women"
                class="c5"
                data-testid="plp-category-quick-link-item-image"
                src="/some/image/src.jpg"
              />
            </div>
            <p
              class="MuiTypography-root c6 MuiTypography-body1"
              data-testid="plp-category-quick-link-item-title"
            >
              Women
            </p>
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            class="c3"
            data-testid="plp-category-quick-link-item"
            href="/women/1"
          >
            <div
              class="c4"
            >
              <img
                alt="women"
                class="c5"
                data-testid="plp-category-quick-link-item-image"
                src="/some/image/src.jpg"
              />
            </div>
            <p
              class="MuiTypography-root c6 MuiTypography-body1"
              data-testid="plp-category-quick-link-item-title"
            >
              Women
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
